File: Pump Fun/typescript/examples/launch.ts
import { JellychainPumpFunSDK, JellychainWallet } from '../src';

async function main() {
  const sdk = new JellychainPumpFunSDK({
    rpcUrl: process.env.SOLANA_RPC_URL || 'https://api.mainnet-beta.solana.com',
    pumpPortalApiKey: process.env.PUMP_PORTAL_API_KEY,
    slippageBps: 1500,
    priorityFeeSol: 0.00005,
  });

  const wallet = process.env.PRIVATE_KEY_BASE58
    ? sdk.walletFromBase58(process.env.PRIVATE_KEY_BASE58)
    : sdk.createWallet();

  console.log('Wallet:', wallet.publicKey.toBase58());
  console.log('Docs:', sdk.docsUrl);

  const result = await sdk.launch({
    wallet: wallet.payer,
    metadata: {
      name: 'Jelly Token',
      symbol: 'JELLY',
      description: 'The official JellyChain memecoin — powered by https://jellychain.fun',
      imageUrl: 'https://picsum.photos/500/500',
      twitter: 'https://twitter.com/jellychain',
      telegram: 'https://t.me/jellychain',
      website: 'https://jellychain.fun',
    },
    devBuyAmountSol: 0.1,
  });

  if (result.success) {
    console.log('Token launched!');
    console.log('Mint address:', result.mintAddress);
    console.log('TX:', result.launchTxSignature);
    console.log('Solscan:', result.solscanUrl);
    console.log('Pump.fun:', result.pumpfunUrl);
  } else {
    console.error('Launch failed:', result.error);
  }
}

main().catch(console.error);

File: Pump Fun/python/examples/launch.py
"""
Example: Launch a token on Pump.fun using JellyChain Python SDK

Docs:     https://jellychain.fun/docs
Terminal: https://terminal.jellychain.fun
GitHub:   https://github.com/jelly-chain
"""
import os
from jellychain_pumpfun import (
    JellychainPumpFunSDK,
    LaunchParams,
    TokenMetadata,
)


def main() -> None:
    sdk = JellychainPumpFunSDK(
        rpc_url=os.getenv("SOLANA_RPC_URL", "https://api.mainnet-beta.solana.com"),
        pump_portal_api_key=os.getenv("PUMP_PORTAL_API_KEY", ""),
        slippage_bps=1500,
        priority_fee_sol=0.00005,
    )

    wallet = sdk.create_wallet() if not os.getenv("PRIVATE_KEY_BASE58") else sdk.wallet_from_base58(os.environ["PRIVATE_KEY_BASE58"])
    print(f"Wallet: {wallet.public_key}")
    print(f"Docs:   {sdk.docs_url}")

    result = sdk.launch(LaunchParams(
        private_key_base58=wallet.to_base58_private_key(),
        metadata=TokenMetadata(
            name="Jelly Token",
            symbol="JELLY",
            description="The official JellyChain memecoin — powered by https://jellychain.fun",
            image_url="https://picsum.photos/500/500",
            twitter="https://twitter.com/jellychain",
            telegram="https://t.me/jellychain",
            website="https://jellychain.fun",
        ),
        dev_buy_amount_sol=0.1,
    ))

    if result.success:
        print("Token launched!")
        print(f"  Mint:    {result.mint_address}")
        print(f"  TX:      {result.launch_tx_signature}")
        print(f"  Solscan: {result.solscan_url}")
        print(f"  Pump.fun:{result.pumpfun_url}")
    else:
        print(f"Launch failed: {result.error}")


if __name__ == "__main__":
    main()
